sklearn/linear_model/glm.py

```diff
@@ -0,0 +1,1813 @@
+    in this case one might say that `y` has a 'scaled' Poisson distributions.
+    The same holds for other distributions.
+
+    Parameters
+    ----------
+    alpha : float, optional (default=1)
+        Constant that multiplies the penalty terms und thus determines the
+        regularization strength.
+        See the notes for the exact mathematical meaning of this
+        parameter.``alpha = 0`` is equivalent to unpenalized GLMs. In this
+        case, the design matrix X must have full column rank
+        (no collinearities).
+
+    l1_ratio : float, optional (default=0)
+        The elastic net mixing parameter, with ``0 <= l1_ratio <= 1``. For
+        ``l1_ratio = 0`` the penalty is an L2 penalty. ``For l1_ratio = 1`` it
+        is an L1 penalty.  For ``0 < l1_ratio < 1``, the penalty is a
+        combination of L1 and L2.
+
+    P1 : {None, array-like}, shape (n_features*,), optional \
+            (default=None)
+        With this array, you can exclude coefficients from the L1 penalty.
+        Set the corresponding value to 1 (include) or 0 (exclude). The
+        default value ``None`` is the same as a 1d array of ones.
+        Note that n_features* = X.shape[1] = length of coef_ (intercept
+        always excluded from counting).
+
+    P2 : {None, array-like, sparse matrix}, shape \
+            (n_features*, n_features*), optional (default=None)
+        With this square matrix the L2 penalty is calculated as `w P2 w`.
+        This gives a fine control over this penalty (Tikhonov
+        regularization).
+        The default value ``None`` is the same as the idendity matrix.
+        Note that n_features* = X.shape[1] = length of coef_ (intercept
+        always excluded from counting). P2 must be positive semi-definite.
+
+    fit_intercept : boolean, optional (default=True)
+        Specifies if a constant (a.k.a. bias or intercept) should be
+        added to the linear predictor (X*coef+intercept).
+
+    family : {'normal', 'poisson', 'gamma', 'inverse.gaussian'} or an instance\
+            of class ExponentialDispersionModel, optional(default='normal')
+        the distributional assumption of the GLM, i.e. which loss function to
+        be minimized.
+
+    link : {'identity', 'log'} or an instance of class Link,
+        optional (default='identity')
+        the link function of the GLM, i.e. mapping from linear predictor
+        (X*coef) to expectation (mu).
+
+    fit_dispersion : {None, 'chisqr', 'deviance'}, optional (defaul=None)
+        method for estimation of the dispersion parameter phi. Whether to use
+        the chi squared statisic or the deviance statistic. If None, the
+        dispersion is not estimated.
+
+    solver : {'auto', 'irls', 'newton-cg', 'lbfgs', 'cd'}, \
+            optional (default='auto')
+        Algorithm to use in the optimization problem.
+
+        - 'auto' sets 'irls' if l1_ratio equals 0, else 'cd'.
+
```

Comment:
these blank lines are not necessary. You could instead use a restructured text definition list.

Reply:
changed to rst text definition list. thanks for the tip.